lib/tasks/hyku_addons/file_set.rake

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+namespace :hyku_addons do
+  namespace :file_set do
+    # Usage: app:hyku_addons:file_set:regenerate_pdf_thumbnails["repo.hyku.docker"]
+    desc "Regenerate thumbnails for works with grayscale thumbnails"
+    task :regenerate_pdf_thumbnails, [:cname] => :environment do |_t, args|
+      AccountElevator.switch!(args[:cname])
+
+      Site.instance.available_works.each do |work_type|
+        work_type = work_type.constantize
+
+        count = work_type.count
+        per_page = 25
+        pages = (count.to_f / per_page.to_f).ceil
+
+        pages.times do |page|
+          work_type.limit(per_page).offset(page * per_page).each do |work|
+            work.file_sets.each do |file_set|
+              next unless file_set.mime_type.match?(/^(.)+\/pdf$/)
+
+              CreateDerivativesJob.set(wait: rand(3600)).perform_later(file_set, file_set.original_file.id)
```

Comment:
I presume you are doing this so as not to overload the job queue and spread the jobs out.  Wouldn't it be better to have a job which spawns the CreateDerivatesJobs for a fixed number of the works which need updating  every hour.  So we'd update say 25 works an hour, and let the workers deal with managing their own loads rather than spreading it out randomly.

Reply:
I did think about it, however simplicity seemed preferable as this might never be used again.